### PR TITLE
Moved from Renderer to Writer

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #1051 

Part 1 of many

@rynowak @NTaylorMullen

### CODE REVIEW

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/CodeGeneration/RuntimeBasicWriterTest.cs [hunk 1]
```diff
@@ -180,6 +180,126 @@ public void WriteCSharpExpression_WithSource_WritesPadding()
...
+            // Act
+            writer.WriteCSharpStatement(context, node);
+
+            // Assert
+            var csharp = context.Writer.Builder.ToString();
+            Assert.Empty(csharp);
+        }
+
+        [Fact]
+        public void WriteCSharpStatement_SkipsLinePragma_withoutSource()
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/CodeGeneration/RuntimeBasicWriterTest.cs]: Without

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/CodeGeneration/RuntimeBasicWriterTest.cs [hunk 2]
```diff
@@ -180,6 +180,126 @@ public void WriteCSharpExpression_WithSource_WritesPadding()
...
+            var csharp = context.Writer.Builder.ToString();
+            Assert.Equal(
+@"if (true) { }
+",
+                csharp,
+                ignoreLineEndingDifferences: true);
+        }
+
+        [Fact]
+        public void WriteCSharpStatement_WritesLinePragma_withSource()
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/CodeGeneration/RuntimeBasicWriterTest.cs]: With

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/CodeGeneration/RuntimeBasicWriterTest.cs [hunk 3]
```diff
@@ -180,6 +180,126 @@ public void WriteCSharpExpression_WithSource_WritesPadding()
                 ignoreLineEndingDifferences: true);
         }
 
+        [Fact]
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/CodeGeneration/RuntimeBasicWriterTest.cs]: Tests for design time?

**Comment by ajaybhargavb** [comment]
🆙 📅 Rebased and added design time tests.

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeBasicWriter.cs [hunk 4]
```diff
@@ -71,7 +72,47 @@ public override void WriteCSharpExpression(CSharpRenderingContext context, CShar
 
         public override void WriteCSharpStatement(CSharpRenderingContext context, CSharpStatementIRNode node)
         {
-            throw new NotImplementedException();
+            var isWhitespaceToken = node.Children.All(child =>
+                child is RazorIRToken token && string.IsNullOrWhiteSpace(token.Content));
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeBasicWriter.cs]: This isn't really the right name, it's more like `isWhitespaceStatement`.

Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeBasicWriter.cs]: Also just do this with a loop, Linq is for bad children

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RuntimeBasicWriter.cs [hunk 5]
```diff
@@ -51,7 +52,40 @@ public override void WriteCSharpExpression(CSharpRenderingContext context, CShar
 
         public override void WriteCSharpStatement(CSharpRenderingContext context, CSharpStatementIRNode node)
         {
-            throw new NotImplementedException();
+            var isWhitespaceToken = node.Children.All(child =>
+                child is RazorIRToken token && string.IsNullOrWhiteSpace(token.Content));
+
+            if (isWhitespaceToken)
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RuntimeBasicWriter.cs]: Same comments here as the last one.

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RuntimeCSharpRenderer.cs [hunk 6]
```diff
@@ -210,6 +210,7 @@ public override void VisitCSharpAttributeValue(CSharpAttributeValueIRNode node)
 
         public override void VisitCSharpStatement(CSharpStatementIRNode node)
         {
+            // We can't remove this yet, because it's still used recursively in a few places.
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RuntimeCSharpRenderer.cs]: Did you validate this? Did you try it? What kinds of things had statements nested inside of them?

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RuntimeCSharpRenderer.cs]: I saw tests fail without this. I didn't really dive in to the details.

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RuntimeCSharpRenderer.cs]: The tests that are failing are all tag helper related. Should be safe to remove once we port tag helper code.

**Pull request merged**